rhp/v3/transport.go

```diff
@@ -95,16 +108,82 @@ type Transport struct {
 	mux *mux.Mux
 }
 
+// stream wraps the mux.Stream type to catch the lazily written subscriber
+// response the host is sending us before the first Read.
+type stream struct {
```

Comment:
Me too. Believe me I tried to use `*mux.Stream` again and manually read the response but the problem with that is that it has to be called precisely before the first `Read`. If you call it before that you end up with a deadlock and after that with a parsing error in whatever code accidentally catches the response.
That said it's not always obvious where this is because depending on how we pay for the RPC, `ProcessPayment` either performs a read or not.